include/seqan3/argument_parser/validators.hpp

```diff
@@ -236,12 +237,17 @@ value_list_validator(std::initializer_list<const char *>) -> value_list_validato
 
 /*!\brief An abstract base class for the file and directory validators.
  * \ingroup argument_parser
+ * \tparam file_t The type of the file to get the valid extensions for; `void` on default.
  *
  * \details
  *
  * This class provides a common interface for seqan3::input_file_validator and the seqan3::output_file_validator as
  * well as the seqan3::input_directory_validator and seqan3::output_directory_validator.
+ *
+ * The type can be further specialised for the seqan3::input_file_validator and the seqan3::output_file_validator
+ * using the template argument to determine the valid extensions from the given file type.
  */
+template <typename file_t = void>
```

Comment:
Because a constructor cannot have an explicit template argument. Thus you would need to pass in an instance of a file. But since it is not default constructible you need to feed it with a stream or file. Yet, it is a weird circular dependency as you need to parse the arguments before you know the filename.

Reply:
Note, that these changes have no implication on the current version, due to type deduction guides. That means, you can provide the file type as a template argument or you call the constructors like before, without any template argument.